tests/components/cast/test_media_player.py

```diff
@@ -358,78 +387,429 @@ async def test_update_cast_chromecasts(hass):
     assert add_dev1.call_count == 1
 
 
+async def test_entity_availability(hass: HomeAssistantType):
+    """Test handling of connection status."""
+    entity_id = "media_player.speaker"
+    info = get_fake_chromecast_info()
+
+    chromecast = await async_setup_media_player_cast(hass, info)
+    _, conn_status_cb, _ = get_status_callbacks(chromecast)
+
+    state = hass.states.get(entity_id)
+    assert state.state == "unavailable"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)